api/org/commcare/api/persistence/SqlHelper.java

```diff
@@ -79,13 +79,14 @@ public static PreparedStatement prepareTableSelectStatement(Connection c,
                                                                 String storageKey,
                                                                 String[] fields,
                                                                 String[] values,
-                                                                Persistable p) {
+                                                                Persistable p) throws IllegalArgumentException{
```

Comment:
I think a good middle-ground is to write a `@throws SomeRuntimeException` javadoc entry for runtime exceptions, instead of adding them to the method signature. Adding a `throws RuntimeException` to a method signature has no semantic effect on how the method or how it is invoked, thus I would argue that it should be avoided and the corresponding information should be moved to comments, where things that don't have semantic effects should live.